.github/workflows/build.yml

```diff
@@ -0,0 +1,137 @@
+    services:
+      postgres:
+        image: postgres:11
+        env:
+          POSTGRES_PASSWORD: ${{ env.POSTGRES_PASSWORD }}
+        options: >-
+          --health-cmd "pg_isready -U postgres"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      graphql-engine:
+        image: hasura/graphql-engine:v1.2.1
+        env:
+          HASURA_GRAPHQL_DATABASE_URL: postgres://postgres:${{ env.POSTGRES_PASSWORD }}@postgres:5432/postgres
+          HASURA_GRAPHQL_ADMIN_SECRET: ${{ env.HASURA_GRAPHQL_ADMIN_SECRET }}
+          ACTION_BASE_ENDPOINT: http://actions-server:3001
+        options: >-
+          --health-cmd "wget --quiet --spider http://localhost:8080/healthz"
+          --health-interval 30s
+          --health-timeout 3s
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # region bootstrap Hasura
+      - name: Apply Hasura migrations
+        uses: tibotiber/hasura-action@master
+        with:
+          args: migrate apply
+        env:
+          HASURA_ENDPOINT: http://graphql-engine:8080
+          HASURA_ADMIN_SECRET: ${{ env.HASURA_GRAPHQL_ADMIN_SECRET }}
+          HASURA_WORKDIR: graphql-server
+      - name: Apply Hasura metadata
+        uses: tibotiber/hasura-action@master
+        with:
+          args: metadata apply
+        env:
+          HASURA_ENDPOINT: http://graphql-engine:8080
+          HASURA_ADMIN_SECRET: ${{ env.HASURA_GRAPHQL_ADMIN_SECRET }}
+          HASURA_WORKDIR: graphql-server
+      # endregion
+
+      # region setup Node.js
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+      - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: "${{ steps.nvm.outputs.NVMRC }}"
+      # endregion
+
+      - name: Install dependencies
+        working-directory: app
+        run: yarn install --frozen-lockfile
+
+      - name: Generate GraphQL code
+        env:
+          REACT_APP_FISHBOWL_GRAPHQL_ENDPOINT: http://graphql-engine:8080/v1/graphql
+          REACT_APP_FISHBOWL_HASURA_ADMIN_SECRET: ${{ env.HASURA_GRAPHQL_ADMIN_SECRET }}
+        working-directory: app
+        run: yarn run gql-gen
+
+      - name: Lint
+        if: github.ref != 'refs/heads/master'
+        working-directory: app
+        run: yarn run lint
+
+      # region build and deploy
+      - name: Build
+        if: github.ref == 'refs/heads/master'
+        env:
+          CI: false
+          REACT_APP_FISHBOWL_GRAPHQL_ENDPOINT: https://fishbowl-graphql.onrender.com/v1/graphql
+          REACT_APP_FISHBOWL_WS_GRAPHQL_ENDPOINT: wss://fishbowl-graphql.onrender.com/v1/graphql
+        working-directory: app
+        run: yarn run build
+      - name: Upload build
+        if: github.ref == 'refs/heads/master'
+        uses: actions/upload-artifact@v2
+        with:
+          name: app
+          path: app/build/
```

Comment:
future: if we were ever to tag this artifact with the sha, we could pull down the artifact for that sha in the deploy script via `RENDER_GIT_COMMIT` https://render.com/docs/environment-variables. 

🤔 through any edge cases that it's fine to keep overwriting "app" each time (e.g. that Render somehow deploys something old etc.)

Reply:
Nice! A similar thought went through my mind, but I didn't find this Render documentation which is certainly relevant.

I'll definitely give this some more thought, but to your point, always deploying the "latest" is probably sufficient for now.

Reply:
It's so hard to google Render documentation... because it's called "render" which means so many other things :facepalm:

Reply:
🤣 I know!